src/main/scala/com/amazon/deequ/checks/Check.scala

```diff
@@ -860,16 +861,120 @@ case class Check(
       includeUpperBound: Boolean = true,
       hint: Option[String] = None)
     : CheckWithLastConstraintFilterable = {
+      isContainedInRange(column, lowerBound, upperBound, includeLowerBound, includeUpperBound, hint)
+  }
 
+  /**
+    * Asserts that the non-null values in a column fall into the predefined interval
+    *
+    * @param column column to run the assertion
+    * @param lowerBound lower bound of the interval
+    * @param upperBound upper bound of the interval
+    * @param includeLowerBound is a value equal to the lower bound allows?
+    * @param includeUpperBound is a value equal to the upper bound allowed?
+    * @param hint A hint to provide additional context why a constraint could have failed
+    * @param quoteBounds should the bound values be quoted?
+    */
+  private def isContainedInRange[T](
+      column: String,
+      lowerBound: T,
+      upperBound: T,
+      includeLowerBound: Boolean,
+      includeUpperBound: Boolean,
+      hint: Option[String] = None,
+      quoteBounds: Boolean = false)
+  : CheckWithLastConstraintFilterable = {
     val leftOperand = if (includeLowerBound) ">=" else ">"
     val rightOperand = if (includeUpperBound) "<=" else "<"
 
-    val predicate = s"$column IS NULL OR " +
-      s"($column $leftOperand $lowerBound AND $column $rightOperand $upperBound)"
+    val predicate = s"$column IS NULL OR " + (
+      if (quoteBounds) {
+        s"($column $leftOperand '$lowerBound' AND $column $rightOperand '$upperBound')"
+      }
+      else {
+        s"($column $leftOperand $lowerBound AND $column $rightOperand $upperBound)"
+      }
+    )
 
     satisfies(predicate, s"$column between $lowerBound and $upperBound", hint = hint)
   }
 
+  /**
+    * Asserts that the non-null values in a timestamp column fall into the predefined interval
+    *
+    * @param column column to run the assertion
+    * @param lowerBound lower bound of the interval
+    * @param upperBound upper bound of the interval
+    * @param includeLowerBound is a value equal to the lower bound allows?
+    * @param includeUpperBound is a value equal to the upper bound allowed?
+    * @param hint A hint to provide additional context why a constraint could have failed
+    */
+  def isContainedIn(
+      column: String,
+      lowerBound: Timestamp,
```

Comment:
I can imagine that some people wouldn't want to define a java.sql.Timestamp. Can they also filter using a string?

For example, in Postgres/Redshift SQL, you can define something like "date >= '2013-01-01" or "date between '2013-01-01' and '2014-01-01'".

How could we support a use case like "the date should not be in the future" or "the date should be within the last 3 days"?

Reply:
Makes sense. Some suggestions:

- we could add a dependency on [scala-time](https://github.com/jorgeortiz85/scala-time/tree/master) / [Joda-time](https://github.com/JodaOrg/joda-time), and accept `DateTime` objects in the `isContainedIn` function. This would allow richer interval specifications directly, such as 

  ```
  isContainedIn("column", DateTime.now(), DateTime.now() - 3.days())  
  ```
  or

  ```
  val now = DateTime.now()
  isContainedIn("column", now, now.minus(Period.days(3)))
  ```

  This would also allow users to specify

  ```
  isContainedIn("column", DateTime.parse("2013-01-01"), DateTime.parse("2014-01-01"))
  ``` 
  which is pretty close to directly supplying `String`s as timestamps.

- in the first version of this PR, I used `java.sql.Timestamp` because of the [direct mapping](https://jaceklaskowski.gitbooks.io/mastering-spark-sql/spark-sql-DataType.html#TimestampType) to Spark's `TimestampType`.

- We could support specification of `String` ranges, in which case the users could directly input bounds such as `"2013-01-01"`. However, such a function could be used for arbitrary `String` ranges, such as `isContainedIn("column", "a", "z")`, which might have unintuitive behavior. If we were to support `String`s as timestamps directly, we might want to provide a dedicated name such as `isContainedInTimeRange`. 


I am favoring the approach to not support date/time Strings but instead provide an interface to `DateTime`, because valid formatting options are explicit by the use of Joda and permissible formats can be directly looked up and are standardized.

Reply:
It makes sense but I have one remark. To support timestamp strings, I recommend to bring in the Joda dependency and let the string-typed method act as a wrapper around the `DateTime`-typed method, like so:

```scala
def isContainedInTimeRange(
    column: String,
    lowerBound: String,
    upperBound: String,
    ...) =
  isContainedInTimeRange(
    column,
    DateTime.parse(lowerBound),
    DateTime.parse(upperBound), 
    ...
  )
```

Reply:
Is that because there is no equivalent nice parse method from the Java SQL API?

One more remark on the parameter names: Should we call them `from` and `to` rather than `lowerBound` and `upperBound`?

Reply:
- Regarding Joda dependency: yes, you are more restricted when using `java.sql.Timestamp`.  Timestamp format must be `yyyy-mm-dd hh:mm:ss[.fffffffff]`, so you cannot simply supply `"2019-01-01"` (which works for Joda's `DateTime`).

- Regarding parameter names: agree, since we will rename to `isContainedInTimeRange`, renaming `lowerBound` to `from` and `upperBound` to `to` seems like a good choice.

Reply:
Okay, let's postpone the Joda import and String variant for now. Maybe we can use it through the JDK8-included Joda/datetime version though?

Reply:
I would prefer the string variant as it is the easiest option for users. We can manually check for strings in the yyyy-mm-dd  format and expand them to the format that Timestamp expects.